manifests/service.pp

```diff
@@ -20,10 +20,43 @@
   $heka_daemon_name = $heka::params::heka_daemon_name
 ) inherits heka::params {
 
-  service { $heka_daemon_name:
-    ensure   => $service_ensure,
-    provider => $service_provider,
-    enable => $service_enable
-  }
+  case $::operatingsystem {
+    'RedHat', 'CentOS': {
+      case $::operatingsystemmajrelease {
+      #Pick Upstart for Red Hat/CentOS 6 and explicitly specify the start, stop, restart
+      #and status commands as a workaround for bugs in the EL6 Upstart provider in older
+      #versions of Puppet 3:
+        '6': {
+          service { $heka_daemon_name:
+            ensure     => $service_ensure,
+            enable     => $service_enable,
```

Comment:
Specifying `enable => true` leads to this error:
```
==> default: Error: Could not enable heka: Execution of '/sbin/chkconfig heka on' returned 1: error reading information on service heka: No such file or directory
```

I think it's trying to use the sysv provider of chkconfig instead of initctl since it doesn't know this is upstart.